app/components/tables/holidays_table/HolidaysTable.tsx

```diff
@@ -0,0 +1,130 @@
+import {
+    useReactTable,
+    flexRender,
+    getCoreRowModel,
+    getPaginationRowModel,
+    getFilteredRowModel,
+    getSortedRowModel,
+} from "@tanstack/react-table";
+import useAxiosPromise from "../../../hooks/useAxiosPromise";
+import React, { useMemo, useState, useEffect } from "react";
+import HolidaysTableHeaders from "@/components/tables/holidays_table/table_element/HolidaysTableHeaders";
+import { Holiday } from "./Holiday";
+import SortArrow from "@/components/tables/holidays_table/table_element/SortArrow";
+import PaginationBar from "@/components/tables/holidays_table/table_element/PaginationBar";
+import FilterBar from "@/components/tables/holidays_table/table_element/FilterBar";
+interface HolidaysTableProps {
+    data: Holiday[];
+}
+
+
+const HolidaysTable = ({ data }: HolidaysTableProps) => {
+    const [filtering, setFiltering] = useState<string>("");
```

Comment:
We can remove the type declaration `<string>` from the `useState` since it is already implied from what's passed into it.